src/application.coffee

```diff
@@ -210,13 +210,15 @@ fetch = (app, setDeviceUpdateState = true) ->
 			throw err
 
 shouldMountKmod = (image) ->
-	Promise.using docker.imageRootDirMounted(image), (rootDir) ->
-		utils.getOSVersion(rootDir + '/etc/os-release')
-	.then (version) ->
-		return version? and (version.match(/^Debian/i) or version.match(/^Raspbian/i))
-	.catch (err) ->
-		console.error('Error getting app OS release: ', err)
-		return false
+	device.getOSVersion().then (osVersion) ->
+		return false  if not osVersion.match(/^Resin OS 1./)
+		Promise.using docker.imageRootDirMounted(image), (rootDir) ->
+			utils.getOSVersion(rootDir + '/etc/os-release')
+		.then (version) ->
+			return version? and (version.match(/^Debian/i) or version.match(/^Raspbian/i))
```

Comment:
It would be nicer to do `/^(Debian|Raspbian)/i.test(version)`:
* the `regex.test()` format as it just returns a boolean directly
* the `/^(Debian|Raspbian)/i` so we only need to one one regex